### PR TITLE
Remove Parcelable implementation for classes with ProductDetails as a member

### PR DESCRIPTION
**Pull request by beylmk**
Because ProductDetails does not have a public constructor, `StoreProduct`, `Package`, `Offering`, `PurchasingData` and `Offerings` are no longer Parcelable.

This PR takes away the implementation/annotation for those classes and updates unit tests accordingly. It also adds a missing test for `StoreProduct`, which caught a failure in parcelization. This is now fixed with the inner `ComparableData` class, copying the pattern we use in `CustomerInfo` to exclude the JSONObject.

### CODE REVIEW

**Review by joshdholtz — APPROVED** [review]
This looks good to me! 💪 LGTM

File: migrations/v6-MIGRATION.md [hunk 1]
```diff
@@ -26,6 +26,7 @@
 | `UpgradeInfo`           | `PurchaseParams.Builder.oldProductId` and `PurchaseParams.Builder.googleProrationMode` |
 | `ProductChangeCallback` | `PurchaseCallback`                                                                     |
 
+`Offering`, `Offerings`, `Package`, and `StoreProduct` are no longer Parcelable.
```
Comment (tonidero) [inline migrations/v6-MIGRATION.md:29]: 👍 I do wonder if this should go to a different section in this doc... Maybe `Breaking changes`?  I think this is ok thoug

File: feature/google/src/test/java/com/revenuecat/purchases/google/BillingWrapperTest.kt [hunk 2]
```diff
@@ -275,7 +275,7 @@ class BillingWrapperTest {
                 fail("shouldn't be an error")
             })
 
-        assertThat(numCallbacks == 1)
+        assert(numCallbacks == 1)
```
Comment (tonidero) [inline feature/google/src/test/java/com/revenuecat/purchases/google/BillingWrapperTest.kt:278]: Hmm I think the pattern we've been following is
```suggestion
        assertThat(numCallbacks).isEqualTo(1)
```

**Review by tonidero — APPROVED** [review]
LGTM!

File: migrations/v6-MIGRATION.md (hunk 1 above)
Comment (beylmk) [reply migrations/v6-MIGRATION.md:29]: josh and i are planning a whole revamp of this doc! stay tuned :)

**Pull request merged**